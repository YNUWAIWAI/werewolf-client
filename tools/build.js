const fs = require('fs')
const path = require('path')
const postcss = require('postcss')
const nested = require('postcss-nested')
const autoprefixer = require('autoprefixer')
const glob = require('glob')
const webpack = require('webpack')
const webpackConfig = require('../webpack.config.js')

const mkdir = dir => {
  return new Promise((resolve, reject) => {
    fs.stat(path.dirname(dir), (err, stats) => {
      if (err) {
        if (err.code === 'ENOENT') {
          fs.mkdir(path.dirname(dir), err => {
            if (err) {
              reject(err)
            }
          })
        } else {
          reject(err)
        }
      }
      resolve()
    })
  })
}

const buildCSS = (src, dest) => {
  fs.readFile(src, (err, css) => {
    if (err) {
      throw err
    }
    postcss([ nested, autoprefixer ])
      .process(css, {
        from: src,
        to: dest
      })
      .then(result => {
        mkdir(dest)
          .then(() => {
            fs.writeFile(dest, result.css, err => {
              if (err) {
                throw err
              }
            })
          })
      })
      .catch(err => {
        console.log(err)
      })
  })
}

const buildHTML = (src, dest) => {
  fs.readFile(src, (err, data) => {
    if (err) {
      throw err
    }
    mkdir(dest)
      .then(() => {
        fs.writeFile(dest, data, err => {
          if (err) {
            throw err
          }
        })
      })
  })
}

const build = destDir => {
  const CSSfiles = glob.sync('src/**/*.css')
  const HTMLfiles = glob.sync('src/**/*.html')

  CSSfiles.forEach(file => {
    buildCSS(file, `${destDir}/stylesheets/${path.relative('src/styles', file)}`)
  })

  HTMLfiles.forEach(file => {
    buildHTML(file, `${destDir}/../app/views/${path.parse(file).name}.scala.html`)
  })
}

const compiler = webpack(webpackConfig)

if (process.argv[2] === '-w') {
  build('../public')
  fs.watch('./src', {recursive: true}, (eventType, filename) => {
    if (filename) {
      console.log(`${eventType}: ${filename}`)
      build('../public')
    }
  })
  compiler.watch({}, (err, stats) => {
    if (err) {
      throw err
    }
  })
} else if (process.argv[2] === '-d') {
  build('docs')
} else {
  build('../public')
  compiler.run((err, stats) => {
    if (err) {
      throw err
    }
    const info = stats.toJson()

    if (stats.hasErrors()) {
      console.error(info.errors)
    }

    if (stats.hasWarnings()) {
      console.warn(info.warnings)
    }
  })
}
