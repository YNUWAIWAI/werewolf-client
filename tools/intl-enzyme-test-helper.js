import {IntlProvider, intlShape} from 'react-intl'
import {mount, shallow} from 'enzyme'
import React from 'react'

export const initRenderer = (locale, messages) => {
  const intlProvider = new IntlProvider(
    {
      locale,
      messages
    },
    {}
  )
  const {intl} = intlProvider.getChildContext()
  const nodeWithIntlProp = node =>
    React.cloneElement(
      node,
      {
        intl
      }
    )

  return {
    mountWithIntl: (node, {context, childContextTypes, ... additionalOptions} = {}) =>
      mount(
        nodeWithIntlProp(node),
        {
          childContextTypes: {
            intl: intlShape,
            ... childContextTypes
          },
          context: {
            ... context,
            intl
          },
          ... additionalOptions
        }
      ),
    shallowWithIntl: (node, {context, ... additionalOptions} = {}) =>
      shallow(
        nodeWithIntlProp(node),
        {
          context: {
            ... context,
            intl
          },
          ... additionalOptions
        }
      )
  }
}

