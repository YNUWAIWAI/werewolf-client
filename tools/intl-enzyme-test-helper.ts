/* eslint no-unused-vars: 0 */
/// <reference types="lobby" />
import * as React from 'react'
import {IntlProvider, intlShape} from 'react-intl'
import {MountRendererProps, ShallowRendererProps, mount, shallow} from 'enzyme'

export const initRenderer = (locale: lobby.Language | village.Language, messages: {[key: string]: string}) => {
  const intlProvider = new IntlProvider(
    {
      locale,
      messages
    },
    {}
  )
  const {intl} = intlProvider.getChildContext()
  const nodeWithIntlProp = (node: React.ReactElement<any>) =>
    React.cloneElement(
      node,
      {
        intl
      }
    )

  return {
    mountWithIntl: (node: React.ReactElement<any>, {context, childContextTypes, ... additionalOptions}: MountRendererProps = {}) =>
      mount(
        nodeWithIntlProp(node),
        {
          childContextTypes: {
            intl: intlShape,
            ... childContextTypes
          },
          context: {
            ... context,
            intl
          },
          ... additionalOptions
        }
      ),
    shallowWithIntl: (node: React.ReactElement<any>, {context, ... additionalOptions}: ShallowRendererProps = {}) =>
      shallow(
        nodeWithIntlProp(node),
        {
          context: {
            ... context,
            intl
          },
          ... additionalOptions
        }
      )
  }
}

