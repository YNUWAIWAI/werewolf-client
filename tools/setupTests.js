const Adapter = require('enzyme-adapter-react-16')
const Enzyme = require('enzyme')
const {JSDOM} = require('jsdom')

Enzyme.configure({
  adapter: new Adapter()
})

const jsdom = new JSDOM(
  '<!doctype html><html><body></body></html>',
  {
    url: 'http://localhost'
  }
)
const {window} = jsdom

const copyProps = (src, target) => {
  Object.defineProperties(target, {
    ... Object.getOwnPropertyDescriptors(src),
    ... Object.getOwnPropertyDescriptors(target)
  })
}

global.window = window
global.document = window.document
global.navigator = {
  userAgent: 'node.js'
}
global.requestAnimationFrame = cb => setTimeout(cb, 0)
global.cancelAnimationFrame = id => {
  clearTimeout(id)
}
copyProps(window, global)
