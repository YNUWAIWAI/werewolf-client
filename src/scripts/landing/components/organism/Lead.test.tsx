import * as React from 'react'
import {
  Language,
  getMessages
} from '../../i18n'
import Lead from './Lead'
import {initRenderer} from '../../tools'

const {mountWithIntl} = initRenderer(Language.en, getMessages(Language.en))

test('render', () => {
  const wrapper = mountWithIntl(
    <Lead />
  )

  expect(wrapper.html()).toMatchSnapshot()
})
