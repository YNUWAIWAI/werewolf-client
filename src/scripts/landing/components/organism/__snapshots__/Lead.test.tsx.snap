// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `"<div class=\\"la--lead\\"><div class=\\"la--title\\"><p>LiCOS: A Game Of Werewolves</p></div><div class=\\"la--flavor\\"><p>* Flavor text *</p></div><iframe allow=\\"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture\\" allowfullscreen=\\"\\" class=\\"la--trailer\\" src=\\"https://www.youtube.com/embed/VrxRKYTRZG8\\"></iframe></div>"`;
